interface/reports/username_report.php

```diff
@@ -0,0 +1,251 @@
+        user_summary();
+        console.log($("#summary").val() + '= summary');
+
+    }
+
+
+});
+
+var iter=0;
+var oTable;
+
+$("#form_from_date").val();
+//Function to initiate datatables plugin
+function init_datatables()
+{
+	oTable=$('#document_table').DataTable({
+        dom: 'Bfrtip',
+        buttons: [
+            'copy', 'excel', 'pdf'
+        ],
+
+        "iDisplayLength": 100,
+        "select":true,
+        "searching":true,
+        "retrieve" : true
+	});
+
+
+
+
+
+	iter=1;
+}
+
+function refreshPage(){
+
+    window.location.reload();
+
+
+
+
+}
+
+//Function to populate document list
+function listusers()
+{
+
+
+    $.ajax({
+        type: "POST",
+        url: "../../library/ajax/username_report_ajax.php",
+        data: {
+            func:"list_all_users",
+            to_date:$("#form_to_date").val(),
+            from_date:$("#form_from_date").val()
+        },
+        beforeSend: function(){
+            $('#image').show();
+        },
+
+        success:function(data)
+        {
+            $('#users_list').html(data);
+            init_datatables();
+        },
+        complete: function(){
+            $('#image').hide();
+        }
+
+    });
+
+
+
+	iter=1;
+}
+
+function user_summary()
+{
+
+    $.ajax({
+        type: "POST",
+        url: "../../library/ajax/username_report_ajax.php",
+        data: {
+            func:"user_summary",
+            to_date: $("#form_to_date").val(),
+            from_date: $("#form_from_date").val()
+        },
+        beforeSend: function(){
+            $('#image').show();
+        },
+
+        success:function(data)
+        {
+            $('#users_list').html(data);
+            init_datatables();
+        },
+        complete: function(){
+            $('#image').hide();
+        }
+
+    });
+
+
+
+
+}
+
+
+
+
+
+</script>
+</head>
+<body class="body_top formtable">&nbsp;&nbsp;
+<form action="./username_report.php" method="post">
+<label><input value="Show Session Details" type="submit" id="details_selector" name="show_all" id="show_all"><?php ?></label>
+
+<label><input value="Show Summary" type="submit" id="summary_selector" name="show_summary" id="show_summary"><?php ?></label>
+<?php
+
+if ( ! $_POST['form_from_date']) {
+    // If a specific patient, default to 2 years ago.
+    $from_date = date("Y-m-d");
+
+} else {
+    $from_date = fixDate($_POST['form_from_date'], date('Y-m-d'));
+    $to_date = fixDate($_POST['form_to_date'], date('Y-m-d'));
+}
+
+if ( !$_POST['form_to_date']) {
+    // If a specific patient, default to 2 years ago.
+    $to_date = date("Y-m-d");
+}
+
+?>
+
+    <input type='text' name='form_from_date' id='form_from_date' size='10'
+           value='<?= $from_date ?>' >
+
+    <input type='text' name='form_to_date' id='form_to_date' size='10'
+           value='<?= $to_date ?>' >
+
+    <input hidden id = 'summary' value = '<?= $_POST['show_summary']  ?>'>
+    <input hidden id = 'details' value = '<?= $_POST['show_all']  ?>'>
```

Comment:
Using the short php tags in the above statements.  Is it needed?

Reply:
Fixed